src/composables/useGoalData.js

```diff
@@ -0,0 +1,127 @@
+import {
+	computed,
+	inject,
+	ref,
+} from 'vue';
+
+import useGoalDataQuery from '#src/graphql/query/useGoalData.graphql';
+import { createUserPreferences, updateUserPreferences } from '#src/util/userPreferenceUtils';
+import postCheckoutAchievementsQuery from '#src/graphql/query/postCheckoutAchievements.graphql';
+
+import {
+	ID_BASIC_NEEDS,
+	ID_CLIMATE_ACTION,
+	ID_REFUGEE_EQUALITY,
+	ID_SUPPORT_ALL,
+	ID_US_ECONOMIC_EQUALITY,
+	ID_WOMENS_EQUALITY,
+} from '#src/composables/useBadgeData';
+
+const GOAL_DISPLAY_MAP = {
+	[ID_BASIC_NEEDS]: 'basic needs loans',
+	[ID_CLIMATE_ACTION]: 'eco-friendly loans',
+	[ID_REFUGEE_EQUALITY]: 'refugees',
+	[ID_SUPPORT_ALL]: 'loans',
+	[ID_US_ECONOMIC_EQUALITY]: 'U.S. entrepreneurs',
+	[ID_WOMENS_EQUALITY]: 'women',
+};
+
+function getGoalDisplayName(category) {
+	return GOAL_DISPLAY_MAP[category] || 'loans';
+}
+
+/**
+ * Vue composable for loading and managing user goal data
+ *
+ * @param {Array} loans - List of loans to count toward goals
+ * @returns Goal data and utilities
+ */
+export default function useGoalData(loans) {
+	const apollo = inject('apollo');
+
+	const loading = ref(true);
+	const currentGoal = ref(null);
+	const userPreferences = ref(null);
+	const overallProgress = ref([]);
+
+	async function loadPreferences(fetchPolicy = 'cache-first') {
+		try {
+			const response = await apollo.query({ query: useGoalDataQuery, fetchPolicy });
+			const prefsData = response.data?.my?.userPreferences || null;
+			userPreferences.value = prefsData;
+			return prefsData ? JSON.parse(prefsData.preferences || '{}') : {};
+		} catch (error) {
+			console.error('Failed to load preferences:', error);
+		}
+	}
+
+	const loadGoalProgress = async () => {
+		try {
+			const loanIds = loans.map(loan => loan.id);
+			const response = await apollo.query({
+				query: postCheckoutAchievementsQuery,
+				variables: { loanIds }
+			});
+			overallProgress.value = response?.data?.postCheckoutAchivements?.overallProgress || [];
+		} catch (error) {
+			console.error('Failed to load goal progress:', error);
+		}
+	};
+
+	function setGoalState(parsedPrefs) {
+		const goals = parsedPrefs.goals || [];
+		currentGoal.value = { ...goals[0] };
+	}
+
+	async function storeGoalPreferences(updates) {
+		if (!userPreferences.value) {
+			await createUserPreferences(apollo, { goals: [] });
+			await loadPreferences('network-only'); // Reload after create
+		}
+		const parsedPrefs = JSON.parse(userPreferences.value.preferences || '{}');
+		const goals = parsedPrefs.goals || [];
+		const goalIndex = goals.findIndex(g => g.goalName === updates.goalName);
+		if (goalIndex !== -1) goals[goalIndex] = { ...goals[goalIndex], ...updates };
+		else goals.push(updates);
+		await updateUserPreferences(apollo, userPreferences.value, parsedPrefs, { goals });
+		setGoalState({ goals }); // Refresh local state after update
+	}
+
+	const goalProgress = computed(() => {
+		const currentProgress = overallProgress.value.find(
```

Comment:
This works for goals that match achievements but not the "all loans" goal. We'll need to also check the loan total for the current user, which should also already be in the TY page query.